apps/studio/pages/api/ai/sql/tools.ts

```diff
@@ -387,5 +387,179 @@ export const getTools = ({
+        3. Do NOT use bare specifiers when importing dependecnies. If you need to use an external dependency, make sure it's prefixed with either \`npm:\` or \`jsr:\`. For example, \`@supabase/supabase-js\` should be written as \`npm:@supabase/supabase-js\`.
+        4. For external imports, always define a version. For example, \`npm:@express\` should be written as \`npm:express@4.18.2\`.
+        5. For external dependencies, importing via \`npm:\` and \`jsr:\` is preferred. Minimize the use of imports from @\`deno.land/x\` , \`esm.sh\` and @\`unpkg.com\` . If you have a package from one of those CDNs, you can replace the CDN hostname with \`npm:\` specifier.
+        6. You can also use Node built-in APIs. You will need to import them using \`node:\` specifier. For example, to import Node process: \`import process from "node:process"\`. Use Node APIs when you find gaps in Deno APIs.
+        7. Do NOT use \`import { serve } from "https://deno.land/std@0.168.0/http/server.ts"\`. Instead use the built-in \`Deno.serve\`.
+        8. Following environment variables (ie. secrets) are pre-populated in both local and hosted Supabase environments. Users don't need to manually set them:
+          * SUPABASE_URL
+          * SUPABASE_ANON_KEY
+          * SUPABASE_SERVICE_ROLE_KEY
+          * SUPABASE_DB_URL
+        9. To set other environment variables (ie. secrets) users can put them in a env file and run the \`supabase secrets set --env-file path/to/env-file\`
+        10. A single Edge Function can handle multiple routes. It is recommended to use a library like Express or Hono to handle the routes as it's easier for developer to understand and maintain. Each route must be prefixed with \`/function-name\` so they are routed correctly.
+        11. File write operations are ONLY permitted on \`/tmp\` directory. You can use either Deno or Node File APIs.
+        12. Use \`EdgeRuntime.waitUntil(promise)\` static method to run long-running tasks in the background without blocking response to a request. Do NOT assume it is available in the request / execution context.
+
+        ## Example Templates
+
+        ### Simple Hello World Function
+
+        \`\`\`edge
+        interface reqPayload {
+          name: string;
+        }
+
+        console.info('server started');
+
+        Deno.serve(async (req: Request) => {
+          const { name }: reqPayload = await req.json();
+          const data = {
+            message: \`Hello \${name} from foo!\`,
+          };
+
+          return new Response(
+            JSON.stringify(data),
+            { headers: { 'Content-Type': 'application/json', 'Connection': 'keep-alive' }}
+          );
+        });
+        \`\`\`
+
+        ### Example Function using Node built-in API
+
+        \`\`\`edge
+        import { randomBytes } from "node:crypto";
+        import { createServer } from "node:http";
+        import process from "node:process";
```

Comment:
@laktek should we suggest importing edge runtime types at the top?

```
// Setup type definitions for built-in Supabase Runtime APIs
import "jsr:@supabase/functions-js/edge-runtime.d.ts";
```

Reply:
yep good call!

Reply:
@SaxonF Can you add this line to all examples here?